code/lib/builder-vite/src/vite-server.ts

```diff
@@ -7,23 +7,24 @@ import { getOptimizeDeps } from './optimizeDeps';
 export async function createViteServer(options: ExtendedOptions, devServer: Server) {
   const { presets } = options;
 
-  const config = await commonConfig(options, 'development');
+  const commonCfg = await commonConfig(options, 'development');
 
-  // Set up dev server
-  config.server = {
-    middlewareMode: true,
-    hmr: {
-      port: options.port,
-      server: devServer,
-    },
-    fs: {
-      strict: true,
+  const config = {
+    ...commonCfg,
+    // Set up dev server
+    server: {
+      middlewareMode: true,
+      hmr: {
+        port: options.port,
+        server: devServer,
+      },
+      fs: {
+        strict: true,
+      },
     },
+    appType: 'custom' as const,
+    optimizeDeps: await getOptimizeDeps(commonCfg, options),
   };
```

Comment:
only minor refactor, same logic.

Reply:
This change means that the `appType` and `server` options are not passed to `getOptimizedDeps()`.  I think this is okay, since we're just passing the config in so that we can create a resolver against which to check the candidates for pre-optimization.

Reply:
good catch, hadn't thought about that.